### ISSUE TITLE
RPC not launching

### ISSUE THREAD

**Issue by TakenMC** [ObserverOfTime/nvimcord#3]
Every time I try to launch the RPC I get this error message `Failed to connect to pipe: \\?\pipe\discord-ipc-0` is there a dependency I am missing or something else.

**Comment by TakenMC** [comment]
![image](https://github.com/ObserverOfTime/nvimcord/assets/68560840/b65a1991-c632-4ec6-aa72-04ca79bb90e1)

**Comment by ObserverOfTime** [comment]
I assume you've started Discord beforehand.

Does it work if you change the `?` to `.` ?
https://github.com/ObserverOfTime/nvimcord/blob/2bdf6183acbddc0e73c97104963a1d2f9b87695b/lua/nvimcord/ipc.lua#L10

**Comment by TakenMC** [comment]
I am not sure how to set that

**Comment by ObserverOfTime** [comment]
I changed it. Update the plugin and see if it works.

**Comment by TakenMC** [comment]
Still the same error
Is there something that i need installed or i just run the plugin

**Comment by ObserverOfTime** [comment]
You need to update the plugin and start Discord before Neovim.

**Comment by TakenMC** [comment]
Still fails to connect but when i listed the path `\\.\pipe\discord-ipc-0` it says it exists

**Comment by ObserverOfTime** [comment]
Are you using Neovim in WSL?

**Comment by TakenMC** [comment]
no

**Comment by cyuria** [comment]
I just switched to this plugin, and I have the same issue. I originally forked the repo to add git integration anyway because it seemed interesting to do, and after playing with this problem, I found that, at least on windows, rpc works regardless of the result of is `IPC.pipe:is_writable()`.
I found that the plugin works perfectly if `IPC:connect()` returns true, that is, it looks like this:
```lua
-- within lua/nvimcord/ipc.lua

---@return boolean
function IPC:connect()
    if not self.pipe or self.pipe:is_closing() then
        self.pipe = assert(vim.loop.new_pipe())
    end

    self.pipe:connect(self.path)

    return true -- changed line
end
```

**Comment by cyuria** [comment]
I've submitted a PR #4 with my git searching additions, as well as a temporary fix for windows which you can have a look at

**Comment by ObserverOfTime** [comment]
Fixed in 3a087d3.
